core/driver/sources/src/main/scala/org/virtuslab/ideprobe/WaitDecision.scala

```diff
@@ -0,0 +1,42 @@
+package org.virtuslab.ideprobe
+
+sealed trait WaitDecision {
+  def needToWait: Boolean
+  def message: Option[String]
+  def ||(other: WaitDecision): WaitDecision = WaitDecision.or(this, other)
+  def &&(other: WaitDecision): WaitDecision = WaitDecision.and(this, other)
```

Comment:
It can be handled with not-that-bad
```
a || b || c
List(x, y, z).reduce(_ || _)
```
if needed, but one would rather want to `or` on the `WaitLogic` than on `WaitDecision`, but `or` and `and` on that class is not `(WaitLogic, WaitLogic) => WaitLogic` function, but `(WaitLogic, ProbeDiver => WaitDecision) => WaitLogic`